src/librustc_data_structures/indexed_set.rs

```diff
@@ -134,11 +131,11 @@ impl<T: Idx> IdxSetBuf<T> {
 
 impl<T: Idx> IdxSet<T> {
     unsafe fn from_slice(s: &[Word]) -> &Self {
-        mem::transmute(s) // (see above WARNING)
+        &*(s as *const [usize] as *const Self)
     }
 
     unsafe fn from_slice_mut(s: &mut [Word]) -> &mut Self {
-        mem::transmute(s) // (see above WARNING)
+        &mut *(s as *mut [usize] as *mut Self)
```

Comment:
This should probably be casting to `*mut [Word]` instead.

Reply:
Good call; I'll update the applicable commit.